### PR TITLE
Fix github download on migration

### PR DESCRIPTION
**Pull request by lunny**
There are two changes on this PR to prevent wrong release download.
- [x] Use a local variable but a point of loop in the closure function
- [x] Refresh rate limit before and after downloading asset from Github

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

**Review by gary-kim — APPROVED** [review]

File: modules/migrations/github.go [hunk 1]
```diff
@@ -311,7 +310,19 @@ func (g *GithubDownloaderV3) convertGithubRelease(rel *github.RepositoryRelease)
 					log.Error("g.client.RateLimits: %s", err)
 				}
 				if asset == nil {
-					return ioutil.NopCloser(bytes.NewBufferString(redir)), nil
+					if redir != "" {
+						g.sleep()
+						resp, err := http.DefaultClient.Get(redir)
```
Comment (zeripath) [inline modules/migrations/github.go]: this DefaultClient is not cancelled when g.ctx is cancelled.

Comment (zeripath) [reply modules/migrations/github.go]:
```suggestion
						req, err := http.NewRequestWithContext(g.ctx, "GET", redir, nil)
						if err != nil {
							return nil, err
						}
						resp, err := http.DefaultClient.Do(req)
```

**Review by zeripath — DISMISSED** [review]
as above

**Review by 6543 — APPROVED** [review]

**Comment by lunny** [comment]
It's unnecessary to backport this PR.

**Pull request merged**